scrapy/extensions/robotstxtparser.py

```diff
@@ -0,0 +1,97 @@
+from six.moves.urllib_robotparser import RobotFileParser
+
+from scrapy.utils.python import to_native_str, to_unicode
+
+class PythonRobotParser():
+    def __init__(self, content):
+        try:
+            content = to_native_str(content)
+        except UnicodeDecodeError:
+            # If we found garbage or robots.txt in an encoding other than UTF-8, disregard it.
+            # Switch to 'allow all' state.
+            content = ''
+        self.rp = RobotFileParser()
+        self.rp.parse(content.splitlines()) 
+
+    def allowed(self, url, useragent):
+        try:
+            useragent = to_native_str(useragent)
+            url = to_native_str(url)
+        except UnicodeDecodeError:
+            return False
```

Comment:
Maybe we should log a warning here.

Reply:
@Gallaecio I think, we should remove the `try/catch` block here. Since if we go by our spec : `user_agent` and `url` have to be strings. Hence `UnicodeDecodeError` can never occur. What do you think?

Reply:
I think you are right.